packages/fuels/tests/harness.rs

```diff
@@ -3596,3 +3602,62 @@ async fn testnet_hello_world() -> Result<(), Error> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_input_message() -> Result<(), Error> {
+    abigen!(
+        MyContract,
+        "packages/fuels/tests/test_projects/contract_test/out/debug/contract_test-abi.json"
+    );
+
+    let compare_messages =
+        |messages_from_provider: Vec<OtherMessage>, used_messages: Vec<Message>| -> bool {
+            zip(&used_messages, &messages_from_provider).all(|(a, b)| {
```

Comment:
nit: I would write it as `iter::zip` here (and `use std::iter`), was confused for a brief moment.

Reply:
ok